bug: rate limit 'Stop and wait' interstitial not detected as degraded session

## Problem

When the Claude CLI hits its rate limit and shows the interactive modal prompt:

```
What do you want to do?
❯ 1. Stop and wait for limit to reset
Enter to confirm · Esc to cancel
```

This blocks forever in automated sessions (no user to press Enter). The agent-wait idle detector eventually kills the session, but the shepherd misclassifies the failure.

## Current Behavior

`_is_degraded_session()` in `base.py` requires **both**:
1. A rate limit warning (`You've used N% of your weekly limit`)
2. Excessive Crystallizing repetitions (>= 5)

The "Stop and wait" modal satisfies neither pattern:
- It doesn't contain the percentage-based warning text
- It doesn't produce Crystallizing output (the session is blocked waiting for input)

As a result, the session falls through to generic "MCP/plugin failure" classification due to status bar noise, masking the real cause.

## Observed Behavior

From 4 consecutive builder failures on issue #2766:

```
# Builder log (runs 3 and 4):
What do you want to do?
❯1.Stopandwaitforlimittoreset
Entertoconfirm·Esctocancel
```

The shepherd logged:
```
[WARN] Builder log shows MCP/plugin failure markers but no git artifacts
[ERROR] builder phase validation failed
```

No mention of rate limiting in the shepherd's error classification.

## Expected Behavior

The shepherd should detect the "Stop and wait for limit to reset" prompt pattern and classify the session as degraded (exit code 11) or a new specific exit code. This should be:
- Detected by `_is_degraded_session()` as an alternative signal alongside Crystallizing
- OR detected by the wrapper's startup monitor as a fatal rate limit condition
- NOT retryable (rate limits don't resolve with retries)

## Suggested Fix

Add "Stop and wait for limit to reset" as an alternative degradation signal in `_is_degraded_session()` that triggers independently of Crystallizing patterns. A session showing this prompt is definitively rate-limited regardless of other signals.

## References

- Observed during shepherd run for #2766
- Degraded session detection: `loom-tools/src/loom_tools/shepherd/phases/base.py:_is_degraded_session()`
- Rate limit detection patterns: `DEGRADED_SESSION_PATTERNS` in same file

--- a/loom-tools/src/loom_tools/shepherd/phases/base.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/base.py
@@ -130,6 +130,15 @@ DEGRADED_SESSION_PATTERNS = [
     re.compile(r"Crystallizing", re.IGNORECASE),
 ]
 
+# "Stop and wait for limit to reset" — the Claude CLI's interactive rate
+# limit modal.  In automated sessions this blocks forever (no user to press
+# Enter).  This is a definitive rate limit signal that classifies as degraded
+# independently of Crystallizing patterns.  The \s* allows for ANSI-stripped
+# output where spaces may be missing.  See issue #2781.
+DEGRADED_STOP_AND_WAIT_PATTERN = re.compile(
+    r"Stop\s*and\s*wait\s*for\s*limit\s*to\s*reset", re.IGNORECASE
+)
+
 # Minimum number of "Crystallizing" occurrences in the last N lines
 # to classify a session as degraded (a single occurrence during normal
 # thinking is not concerning).
@@ -882,25 +891,19 @@ def _is_ghost_session(
 
 
 def _is_degraded_session(log_path: Path) -> bool:
-    """Check if a session log indicates a degraded session (rate limits, Crystallizing loops).
+    """Check if a session log indicates a degraded session due to rate limits.
 
     A degraded session is one where the builder ran but produced no useful work
-    because it was resource-constrained.  This manifests as:
-    - Rate limit warnings ("You've used 87% of your weekly limit")
-    - Repeated "Crystallizing..." output (Claude's extended thinking under pressure)
+    because it was resource-constrained.  This manifests as either:
 
-    Degraded sessions are distinct from low-output sessions (which never started)
-    and stuck sessions (which started but stopped making progress).  A degraded
-    session actively produces output but the output is non-productive.
+    Path A — "Stop and wait" modal (standalone, see issue #2781):
+      The CLI shows an interactive rate limit prompt that blocks forever in
+      automated sessions.  This is a definitive rate limit signal.
 
-    Detection requires BOTH:
-    1. A rate limit warning in the log, AND
-    2. Excessive "Crystallizing" repetitions (>= DEGRADED_CRYSTALLIZING_THRESHOLD)
-
-    This two-signal approach prevents false positives: a single rate limit warning
-    during an otherwise productive session is harmless, and brief "Crystallizing"
-    during normal thinking is expected.  The combination indicates the session has
-    entered a non-recoverable degraded state.  See issue #2631.
+    Path B — Rate limit warning + Crystallizing loop (see issue #2631):
+      Rate limit warnings ("You've used 87% of your weekly limit") combined
+      with repeated "Crystallizing..." output.  Requires BOTH signals to
+      prevent false positives.
 
     Args:
         log_path: Path to the worker session log file.
@@ -918,12 +921,21 @@ def _is_degraded_session(log_path: Path) -> bool:
         if not cli_output:
             return False
 
-        # Signal 1: Rate limit warning present
+        # Path A: "Stop and wait for limit to reset" modal (definitive,
+        # standalone).  The CLI is blocked waiting for user input that will
+        # never come in an automated session.  See issue #2781.
+        if DEGRADED_STOP_AND_WAIT_PATTERN.search(cli_output):
+            return True
+
+        # Path B: Rate limit warning + Crystallizing (two-signal detection).
+        # A single rate limit warning during an otherwise productive session
+        # is harmless, and brief "Crystallizing" during normal thinking is
+        # expected.  The combination indicates a non-recoverable degraded
+        # state.  See issue #2631.
         has_rate_limit = bool(DEGRADED_SESSION_PATTERNS[0].search(cli_output))
         if not has_rate_limit:
             return False
 
-        # Signal 2: Excessive Crystallizing repetitions
         lines = cli_output.splitlines()
         tail = lines[-DEGRADED_SCAN_TAIL_LINES:]
         crystallizing_count = sum(
@@ -942,8 +954,9 @@ def _scan_log_for_degradation(log_path: Path) -> bool:
     for in-flight polling during ``run_worker_phase()``.  It reads only the
     tail of the file to minimize I/O overhead.
 
-    Returns True if the log shows both rate limit warnings and excessive
-    "Crystallizing" repetitions.  See issue #2631.
+    Returns True if the log shows the "Stop and wait" rate limit modal
+    (standalone) or both rate limit warnings and excessive "Crystallizing"
+    repetitions.  See issues #2631, #2781.
     """
     if not log_path.is_file():
         return False
@@ -963,6 +976,11 @@ def _scan_log_for_degradation(log_path: Path) -> bool:
         tail = lines[-DEGRADED_SCAN_TAIL_LINES:]
         text = "\n".join(tail)
 
+        # Path A: "Stop and wait" modal (standalone).  See issue #2781.
+        if DEGRADED_STOP_AND_WAIT_PATTERN.search(text):
+            return True
+
+        # Path B: Rate limit warning + Crystallizing.  See issue #2631.
         has_rate_limit = bool(DEGRADED_SESSION_PATTERNS[0].search(text))
         if not has_rate_limit:
             return False
@@ -1288,7 +1306,7 @@ def run_worker_phase(
         - 8: Planning stall detected (stuck in planning checkpoint)
         - 9: Auth pre-flight failure (not retryable, see issue #2508)
         - 10: Ghost session detected (instant exit, no work — see issue #2604)
-        - 11: Degraded session detected (rate limits + Crystallizing loop, see issue #2631)
+        - 11: Degraded session detected (rate limits, see issues #2631, #2781)
         - Other: Error
     """
     # Use unique session name per retry attempt to prevent tmux state
@@ -1428,9 +1446,9 @@ def run_worker_phase(
     # If it stays there beyond planning_timeout, terminate the worker.
     _planning_first_seen: float | None = None
 
-    # Degraded session detection state (issue #2631).
-    # Periodically scan the log for rate limit + Crystallizing patterns
-    # and abort early rather than waiting for the full timeout.
+    # Degraded session detection state (issues #2631, #2781).
+    # Periodically scan the log for rate limit patterns and abort early
+    # rather than waiting for the full timeout.
     _last_degraded_scan: float = time.monotonic()
 
     while wait_proc.poll() is None:
@@ -1478,10 +1496,11 @@ def run_worker_phase(
                 # Checkpoint advanced past planning (or doesn't exist yet)
                 _planning_first_seen = None
 
-        # In-flight degraded session detection (issue #2631).
-        # Periodically scan the log for rate limit + Crystallizing patterns.
-        # Abort early to avoid wasting the entire builder time budget on a
-        # session that will never produce useful output.
+        # In-flight degraded session detection (issues #2631, #2781).
+        # Periodically scan the log for rate limit patterns (Crystallizing
+        # loop or "Stop and wait" modal).  Abort early to avoid wasting the
+        # entire builder time budget on a session that will never produce
+        # useful output.
         degraded_elapsed = time.monotonic() - _last_degraded_scan
         if degraded_elapsed >= _DEGRADED_SCAN_INTERVAL:
             _last_degraded_scan = time.monotonic()
@@ -1489,7 +1508,7 @@ def run_worker_phase(
             if _scan_log_for_degradation(degraded_log_path):
                 log_warning(
                     f"Degraded session detected: {role} session '{actual_name}' "
-                    f"shows rate limit warnings and Crystallizing loop, "
+                    f"shows rate limit indicators, "
                     f"terminating early (log: {degraded_log_path})"
                 )
                 wait_proc.terminate()
@@ -1612,18 +1631,19 @@ def run_worker_phase(
         log_warning(f"Post-mortem: {postmortem['summary']}")
         return 10
 
-    # Check for degraded session (exit code 11) — rate limits + Crystallizing.
-    # Degraded sessions produce output but it's non-productive garbage.
+    # Check for degraded session (exit code 11) — rate limit indicators.
+    # Degraded sessions produce output but it's non-productive garbage,
+    # or block forever on a rate limit modal prompt.
     # This is NOT retryable: the underlying resource constraint (rate limits)
     # will persist until the limit resets.  Check before MCP/low-output
     # because degraded sessions may have substantial output volume.
-    # See issue #2631.
+    # See issues #2631, #2781.
     if _is_degraded_session(log_path):
         if postmortem is not None:
             log_warning(f"Post-mortem: {postmortem['summary']}")
         log_warning(
             f"Degraded session detected for {role} session '{actual_name}': "
-            f"rate limit warnings and Crystallizing loop "
+            f"rate limit indicators detected "
             f"(exit code {wait_exit}, log: {log_path})"
         )
         return 11
@@ -1751,9 +1771,9 @@ def run_phase_with_retry(
 
         # --- Degraded session (exit code 11) ---
         # Not retryable: the builder session was degraded by rate limits
-        # and entered a Crystallizing loop.  The underlying rate limit
-        # won't resolve with retries — the daemon should back off or
-        # pick a different issue.  See issue #2631.
+        # (Crystallizing loop or "Stop and wait" modal).  The underlying
+        # rate limit won't resolve with retries — the daemon should back
+        # off or pick a different issue.  See issues #2631, #2781.
         if exit_code == 11:
             return 11
 

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -498,20 +498,19 @@ class BuilderPhase:
             )
 
         if exit_code == 11:
-            # Degraded session: builder ran under rate limits and entered a
-            # Crystallizing loop, producing no useful work.  Not retryable —
-            # the rate limit won't resolve until it resets.  See issue #2631.
+            # Degraded session: builder hit rate limits (Crystallizing loop
+            # or "Stop and wait" modal).  Not retryable — the rate limit
+            # won't resolve until it resets.  See issues #2631, #2781.
             log_warning(
                 f"Degraded session for issue #{ctx.config.issue}: "
-                f"builder was rate-limited and entered Crystallizing loop"
+                f"builder was rate-limited"
             )
             self._cleanup_stale_worktree(ctx)
             return PhaseResult(
                 status=PhaseStatus.FAILED,
                 message=(
-                    "builder session degraded: rate limit warnings and "
-                    "Crystallizing loop detected (not retryable until "
-                    "rate limit resets)"
+                    "builder session degraded: rate limit detected "
+                    "(not retryable until rate limit resets)"
                 ),
                 phase_name="builder",
                 data={

--- a/loom-tools/tests/shepherd/test_degraded_session_detection.py
+++ b/loom-tools/tests/shepherd/test_degraded_session_detection.py
@@ -1,9 +1,16 @@
-"""Tests for degraded session detection (issue #2631).
+"""Tests for degraded session detection (issues #2631, #2781).
 
-Tests the detection of builder sessions running under rate limits that enter
-a non-productive Crystallizing loop.  Detection requires BOTH:
-1. A rate limit warning ("You've used X% of your weekly limit")
-2. Excessive "Crystallizing..." repetitions (>= threshold)
+Tests the detection of builder sessions degraded by rate limits.  Two detection
+paths:
+
+Path A — "Stop and wait for limit to reset" modal (standalone, issue #2781):
+  The CLI shows an interactive rate limit prompt that blocks forever in
+  automated sessions.  This is a definitive rate limit signal.
+
+Path B — Rate limit warning + Crystallizing loop (issue #2631):
+  Requires BOTH:
+  1. A rate limit warning ("You've used X% of your weekly limit")
+  2. Excessive "Crystallizing..." repetitions (>= threshold)
 """
 
 from __future__ import annotations
@@ -19,6 +26,7 @@ from loom_tools.shepherd.context import ShepherdContext
 from loom_tools.shepherd.phases.base import (
     DEGRADED_CRYSTALLIZING_THRESHOLD,
     DEGRADED_SCAN_TAIL_LINES,
+    DEGRADED_STOP_AND_WAIT_PATTERN,
     _is_degraded_session,
     _scan_log_for_degradation,
 )
@@ -124,6 +132,43 @@ class TestIsDegradedSession:
             _write_log(tmp_log, _make_degraded_log(rate_limit_pct=pct))
             assert _is_degraded_session(tmp_log) is True, f"Failed for {pct}%"
 
+    # --- "Stop and wait" modal detection (issue #2781) ---
+
+    def test_degraded_with_stop_and_wait_modal(self, tmp_log: Path) -> None:
+        """'Stop and wait for limit to reset' modal → degraded (standalone)."""
+        content = (
+            "What do you want to do?\n"
+            "❯ 1. Stop and wait for limit to reset\n"
+            "Enter to confirm · Esc to cancel\n"
+        )
+        _write_log(tmp_log, content)
+        assert _is_degraded_session(tmp_log) is True
+
+    def test_degraded_with_stop_and_wait_no_spaces(self, tmp_log: Path) -> None:
+        """ANSI-stripped 'Stopandwaitforlimittoreset' → degraded."""
+        content = (
+            "Whatdoyouwanttodo?\n"
+            "❯1.Stopandwaitforlimittoreset\n"
+            "Entertoconfirm·Esctocancel\n"
+        )
+        _write_log(tmp_log, content)
+        assert _is_degraded_session(tmp_log) is True
+
+    def test_stop_and_wait_without_crystallizing(self, tmp_log: Path) -> None:
+        """'Stop and wait' modal needs no Crystallizing to detect."""
+        content = (
+            "Some normal output\n"
+            "❯ 1. Stop and wait for limit to reset\n"
+        )
+        _write_log(tmp_log, content)
+        assert _is_degraded_session(tmp_log) is True
+
+    def test_stop_and_wait_without_rate_limit_warning(self, tmp_log: Path) -> None:
+        """'Stop and wait' needs no percentage-based rate limit warning."""
+        content = "Stop and wait for limit to reset\n"
+        _write_log(tmp_log, content)
+        assert _is_degraded_session(tmp_log) is True
+
 
 # ---------------------------------------------------------------------------
 # _scan_log_for_degradation tests (in-flight polling variant)
@@ -152,6 +197,23 @@ class TestScanLogForDegradation:
     def test_handles_missing_file(self, tmp_log: Path) -> None:
         """Missing file returns False without error."""
         assert _scan_log_for_degradation(tmp_log) is False
+
+    def test_detects_stop_and_wait_modal(self, tmp_log: Path) -> None:
+        """Detects 'Stop and wait' rate limit modal during polling."""
+        content = (
+            "Working on feature...\n"
+            "What do you want to do?\n"
+            "❯ 1. Stop and wait for limit to reset\n"
+            "Enter to confirm · Esc to cancel\n"
+        )
+        _write_log(tmp_log, content)
+        assert _scan_log_for_degradation(tmp_log) is True
+
+    def test_detects_stop_and_wait_no_spaces(self, tmp_log: Path) -> None:
+        """Detects ANSI-stripped 'Stopandwaitforlimittoreset' during polling."""
+        content = "❯1.Stopandwaitforlimittoreset\n"
+        _write_log(tmp_log, content)
+        assert _scan_log_for_degradation(tmp_log) is True
 
 
 # ---------------------------------------------------------------------------
@@ -214,4 +276,3 @@ class TestBuilderDegradedSessionHandling:
         assert result.data.get("degraded_session") is True
         assert result.data.get("exit_code") == 11
         assert "rate limit" in result.message.lower()
-        assert "crystallizing" in result.message.lower()

